### PR TITLE
Fix declarations after statements

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR updates the codebase to remove declarations after statements that are non compliant as per the C90 standard.

Test Steps
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- ~[ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.~

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -217,9 +217,6 @@ add_compile_options(
     $<$<COMPILE_LANG_AND_ID:C,Clang>:-Wno-extra-semi-stmt>
     $<$<COMPILE_LANG_AND_ID:C,Clang>:-Wno-missing-noreturn>
     $<$<COMPILE_LANG_AND_ID:C,Clang>:-Wno-cast-align>
-
-    # Suppressions required to build clean with GCC.
-    $<$<COMPILE_LANG_AND_ID:C,GNU>:-Wno-declaration-after-statement>
```
Comment (moninom1) [inline CMakeLists.txt:222]: If we are removing this have you checked build with gcc

Comment (tony-josi-aws) [reply CMakeLists.txt:222]: Yes, the C90 compliance checks has been done on GCC and Clang

**Review by shubnil — COMMENTED** [review]
The change looks good.
But, I have 2 queries.
1. Is declaration at the top enough for C90 or should we initialize to some default values as well
2. Are we just moving the declaration up and initialization is still at its place

**Comment by tony-josi-aws** [comment]
1. Declaration itself is enough
2. If we are able to know the data before hand like constants we are initializing it during declaration itself, but for the cases were it depends on other variables/function calls we defer the initialization until other variables are known

**Review by shubnil — APPROVED** [review]

**Review by kar-rahul-aws — APPROVED** [review]

**Pull request merged**